### DISCUSSION TITLE
password with "#"

### DISCUSSION THREAD

**Discussion by dbaid** [sijms/go-ora#215] [General]
Hello,

     parse error, would you mind telling me how to solve it ?
     password is "aaa#bbb"

oracle://test:aaa#bbb@test:1521/test
panic: error pinging db: parse "oracle://test:aaa": invalid port ":aaa" after host

**Comment by sijms** [comment]
use function go_ora.BuildUrl()

**Reply by dbaid** [reply]
Solved!! Thank you~
Regards,
Steven

package main

import (
        "database/sql"
        "fmt"
        "os"

        go_ora "github.com/sijms/go-ora/v2"
)

func main() {
        urlOptions := map[string]string{
                "CONNECTION TIMEOUT": "3",
        }
        databaseUrl := go_ora.BuildUrl("localhost", 1521, "XE", "system", "pass#aaa", urlOptions)
        conn, err := sql.Open("oracle", databaseUrl)
        if err != nil {
                fmt.Println(err)
                os.Exit(1)
        }
        err = conn.Ping()
        if err != nil {
                fmt.Println(err)
                return
        }
        fmt.Println("aaa")
}

C:\Users\User\go\src\test2>go run .
aaa
